site/src/pages/DeploySettingsPage/NotificationsPage/NotificationsPage.stories.tsx

```diff
@@ -0,0 +1,258 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { spyOn, userEvent, within } from "@storybook/test";
+import { API } from "api/api";
+import {
+  notificationDispatchMethodsKey,
+  systemNotificationTemplatesKey,
+} from "api/queries/notifications";
+import type { DeploymentValues, SerpentOption } from "api/typesGenerated";
+import {
+  MockNotificationMethodsResponse,
+  MockNotificationTemplates,
+  MockUser,
+} from "testHelpers/entities";
+import {
+  withAuthProvider,
+  withDashboardProvider,
+  withDeploySettings,
+  withGlobalSnackbar,
+} from "testHelpers/storybook";
+import { NotificationsPage } from "./NotificationsPage";
+
+const meta: Meta<typeof NotificationsPage> = {
+  title: "pages/DeploymentSettings/NotificationsPage",
+  component: NotificationsPage,
+  parameters: {
+    experiments: ["notifications"],
+    queries: [
+      { key: systemNotificationTemplatesKey, data: MockNotificationTemplates },
+      {
+        key: notificationDispatchMethodsKey,
+        data: MockNotificationMethodsResponse,
+      },
+    ],
+    user: MockUser,
+    permissions: { viewDeploymentValues: true },
+    deploymentOptions: mockNotificationOptions(),
+    deploymentValues: {
+      notifications: {
+        webhook: {
+          endpoint: "https://example.com",
+        },
+      },
+    } as DeploymentValues,
+  },
+  decorators: [
+    withGlobalSnackbar,
+    withAuthProvider,
+    withDashboardProvider,
+    withDeploySettings,
+  ],
+};
+
+export default meta;
+
+type Story = StoryObj<typeof NotificationsPage>;
+
+export const Default: Story = {};
+
+export const NoWebhookEndpoint: Story = {
+  parameters: {
+    deploymentValues: {
+      notifications: {
+        webhook: {
+          endpoint: "",
+        },
+      },
+    } as DeploymentValues,
+  },
+};
+
+export const Toggle: Story = {
+  play: async ({ canvasElement }) => {
+    spyOn(API, "updateNotificationTemplateMethod").mockResolvedValue();
+    const user = userEvent.setup();
+    const canvas = within(canvasElement);
+    const option = await canvas.findByText("Workspace Marked as Dormant");
+    const toggleButton = within(option.closest("li")!).getByRole("button", {
+      name: "Webhook",
+    });
+    await user.click(toggleButton);
+  },
+};
+
+export const Settings: Story = {
+  play: async ({ canvasElement }) => {
+    const user = userEvent.setup();
+    const canvas = within(canvasElement);
+    const settingsTab = await canvas.findByText("Settings");
+    await user.click(settingsTab);
+  },
+};
+
+function mockNotificationOptions(): SerpentOption[] {
```

Comment:
Potential follow-up: we could generate these automatically and not need to worry about keeping them up to date!